travis-ci/managers/ubuntu-build-check.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+set -e
+
+info() { echo -e "\033[33;1m$1\033[0m"; }
+error() { echo >&2 -e "\033[31;1m$1\033[0m"; }
+success() { echo >&2 -e "\033[32;1m$1\033[0m"; }
+
+ARGS=(
+    "--optimization=0"
+    "--optimization=2"
+    "--optimization=3"
+    "--optimization=s"
+    "-Db_lto=true"
+    "-Db_ndebug=true"
+)
+PACKAGES=(
+    cryptsetup-bin
+    expect
+    fdisk
+    gettext
+    iptables-dev
+    iputils-ping
+    isc-dhcp-client
+    itstool
+    kbd
+    libblkid-dev
+    libcap-dev
+    libcurl4-gnutls-dev
+    libfdisk-dev
+    libgpg-error-dev
+    liblz4-dev
+    liblzma-dev
+    libmicrohttpd-dev
+    libmount-dev
+    libp11-kit-dev
+    libpwquality-dev
+    libqrencode-dev
+    libssl-dev
+    libxkbcommon-dev
+    libzstd-dev
+    mount
+    net-tools
+    ninja-build
+    perl
+    python-lxml
+    python3-evdev
+    python3-lxml
+    python3-pip
+    python3-pyparsing
+    python3-setuptools
+    quota
+    strace
+    unifont
+    util-linux
+    zstd
+)
+CC="${CC:-gcc}"
+CXX="${CXX:-g++}"
+AR="${AR:-""}"
+RELEASE="${1:-bionic}"
+
+bash -c "echo 'deb-src http://archive.ubuntu.com/ubuntu/ $RELEASE main restricted universe multiverse' >>/etc/apt/sources.list"
+
+apt-get update
+apt-get build-dep systemd -y
+apt-get install -y "${PACKAGES[@]}"
+# Install latest meson from pip, as the distro-one doesn't support
+# --optimization=
+pip3 install meson
+
+$CC --version
+
+# Store the build logs in a predictable place so the artifact collecton
+# from GH actions can pick them up
+rm -fr _logs
+mkdir _logs
+
+for args in "${ARGS[@]}"; do
+    SECONDS=0
+    # Sanitize the $args string so it can be used as a filename
+    LOG_SUFFIX="$(echo $args | tr -cd '[:alnum:]._-').log"
+    MESON_LOG="_logs/meson_$LOG_SUFFIX"
+    NINJA_LOG="_logs/ninja_$LOG_SUFFIX"
+
+    info "Checking build with $args"
+    if ! AR="$AR" CC="$CC" CXX="$CXX" meson --werror $args build &> "$MESON_LOG"; then
+        cat "$MESON_LOG"
+        error "meson failed with $args"
+        exit 1
+    fi
+
+    if ! ninja -v -C build &> "$NINJA_LOG"; then
+        cat "$NINJA_LOG"
+        error "ninja failed with $args"
+        exit 1
+    fi
+
+    rm -fr build
```

Comment:
I think to start with a clean slate we should use something like `git clean -dxf` here.